01-main/packages/onefetch

```diff
@@ -0,0 +1,9 @@
+DEFVER=1
+get_github_releases "o2sh/onefetch" "latest"
+if [ "${ACTION}" != "prettylist" ]; then
+    URL=$(grep "browser_download_url.*\.deb\"" "${CACHE_FILE}" | head -n1 | cut -d'"' -f4)
```

Comment:
Per https://stackoverflow.com/a/17086464 we can extract the URL with a single command:
`URL=$(sed -E -n '/^.*browser_download_url.*"([^"].+\.deb)".*/{s//\1/p;q}' "${CACHE_FILE}")` 
Though it's admittedly more complex.
Also
`VERSION_PUBLISHED="$(echo "${URL}" | cut -d'/' -f8)"`
can be shortened to
`VERSION_PUBLISHED=$(cut -d'/' -f8 <<<"${URL}")`
Note the enclosing quotes are never needed when executing a command.

Reply:
More complex, more painful to parse at 2am, harder to maintain 2 years later when upstream introduces RiscV and Arm debs or github api gets a subtle rev, and most importantly not helpful to the first time potential contributor just trying to do something entertaining as the their first foray into `bash` copy-pasta programming.

```
real	0m0.004s
user	0m0.004s
sys	    0m0.000s
```
is a significant improvement on
```
real	0m0.009s
user	0m0.007s
sys	    0m0.007s
```
but hardly time to make a cuppa so I'll stick with readability here - but thanks for the `sed` trick ! TIL

Reply:
Thanks @rasa . I very much appreciate and value your helpful review. :star:

Reply:
You might consider simplifying to
`URL=$(grep -m 1 "browser_download_url.*\.deb\"" "${CACHE_FILE}" | cut -d'"' -f4)`
as it's faster, removes complexity, and doesn't obfuscate. But no offense taken if you're not keen on this sort of optimization.

Reply:
I love optimizations.  This is clear and I will adopt it.
Did you know you can make this kind of suggestion directly in a review comment (click a + on the line and then click the first button above the comment or type <ctrl-g> or directly enter with markdown

  ```suggestion
  URL=$(grep -m 1 "browser_download_url.*\.deb\"" "${CACHE_FILE}" | cut -d'"' -f4)
  ```
This allows the author/maintainer to easily adopt and commit your suggestion(s) and I think ensures you get a credit iirc.